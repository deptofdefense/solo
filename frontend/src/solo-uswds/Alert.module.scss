.extWidth {
    width: 100vw;
}