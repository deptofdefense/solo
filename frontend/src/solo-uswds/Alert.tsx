import React from "react";
import classNames from "classnames";

interface AlertProps extends Partial<JSX.IntrinsicElements["div"]> {
  status: "success" | "warning" | "error" | "info";
  heading?: string;
  text?: string;
}

export const Alert: React.FC<AlertProps> = ({
  status,
  heading,
  text,
  className,
  ...rest
}) => (
  <div
    className={classNames("usa-alert", `usa-alert--${status}`, className)}
    {...rest}
  >
    <div className="usa-alert__body">
      {heading && <h3 className="usa-alert__heading">{heading}</h3>}
      {text && <p className="usa-alert__text">{text}</p>}
    </div>
  </div>
);
