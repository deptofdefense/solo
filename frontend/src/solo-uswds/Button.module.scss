.square {
  border-radius: 0 !important;
}
