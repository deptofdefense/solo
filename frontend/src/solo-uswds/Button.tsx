import React, { useState } from "react";
import classNames from "classnames";

interface ButtonProps {
  focused?: boolean;
  active?: boolean;
  disabled?: boolean;
  outline?: boolean;
  inverse?: boolean;
  big?: boolean;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  color?: "secondary" | "accent-cool" | "base";
  className?: string;
}

export const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  color,
  focused = false,
  active = false,
  outline = false,
  inverse = false,
  big = false,
  disabled = false,
  className = ""
}) => {
  const [isHovered, setIsHovered] = useState(false);

  return (
    <button
      onClick={onClick}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
      disabled={disabled}
      className={classNames(className, "usa-button", {
        "usa-button--hover": isHovered,
        "usa-button--active": active,
        "usa-button--outline": outline,
        "usa-button--inverse": inverse,
        "usa-focus": focused,
        "usa-button--secondary": color === "secondary",
        "usa-button--accent-cool": color === "accent-cool",
        "usa-button--base": color === "base",
        "usa-button--big": big
      })}
    >
      {children}
    </button>
  );
};
