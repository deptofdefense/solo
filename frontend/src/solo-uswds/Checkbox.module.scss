.inp {
  position: unset !important;
  left: 0 !important;
}
