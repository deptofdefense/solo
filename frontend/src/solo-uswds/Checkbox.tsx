import React from "react";
import classNames from "classnames";
import classes from "./Checkbox.module.scss";

interface CheckboxProps extends Partial<JSX.IntrinsicElements["input"]> {
  checked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
}

export const Checkbox: React.FC<CheckboxProps> = ({
  checked,
  onChange,
  className,
  ...rest
}) => (
  <input
    className={classNames("usa-checkbox__input", classes.inp, className)}
    onChange={onChange}
    type="checkbox"
    checked={checked}
    {...rest}
  />
);
