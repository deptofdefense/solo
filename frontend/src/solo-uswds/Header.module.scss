.navContainer {
  max-width: none !important;
  margin-top: 2rem;
  align-items: flex-start !important;
}
@media only screen and (max-width: 64em) {
  div.navContainer {
    margin-top: 0rem;
  }
}