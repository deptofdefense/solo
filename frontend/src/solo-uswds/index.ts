export { Button } from "./Button";
export { OfficialSiteBanner } from "./OfficialSiteBanner";
export { Header, HeaderNavbar } from "./Header";
export { HeaderNav } from "./HeaderNav";
export { HeaderNavLink } from "./HeaderNavLink";
export { HeaderLogo } from "./HeaderLogo";
