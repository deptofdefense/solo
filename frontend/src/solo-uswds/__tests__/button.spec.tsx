import React from "react";
import { render, wait, fireEvent } from "test-utils";
import { Button } from "solo-uswds";

describe("uswds button component", () => {
  const clickMock = jest.fn();

  afterEach(() => {
    clickMock.mockReset();
  });

  it("matches snapshot", () => {
    const { asFragment, getByText } = render(
      <Button onClick={clickMock}>testbtn</Button>
    );
    const btn = getByText("testbtn");
    expect(btn).toHaveClass("usa-button");
    expect(btn).not.toHaveClass(
      "usa-button--outline",
      "usa-button--active",
      "usa-button--inverse",
      "usa-focus",
      "usa-button--big"
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds other classes based on props", () => {
    const { getByText } = render(
      <Button
        onClick={clickMock}
        focused
        outline
        active
        inverse
        big
        className="someotherclassfromparent"
      >
        testbtn
      </Button>
    );
    const btn = getByText("testbtn");
    expect(btn).toHaveClass(
      "usa-button",
      "usa-button--outline",
      "usa-button--active",
      "usa-button--inverse",
      "usa-focus",
      "usa-button--big"
    );
  });

  it("calls click handler on click", () => {
    const { getByText } = render(<Button onClick={clickMock}>clickme</Button>);
    const btn = getByText("clickme");
    fireEvent.click(btn);
    expect(clickMock).toHaveBeenCalled();
  });

  it("adds appropriate secondary color class", () => {
    const { getByText } = render(
      <Button onClick={clickMock} color="secondary">
        testbtn
      </Button>
    );
    const btn = getByText("testbtn");
    expect(btn).toHaveClass("usa-button--secondary");
    expect(btn).not.toHaveClass("usa-button--base");
  });

  it("adds appropriate cool accent color class", () => {
    const { getByText } = render(
      <Button onClick={clickMock} color="accent-cool">
        testbtn
      </Button>
    );
    const btn = getByText("testbtn");
    expect(btn).toHaveClass("usa-button--accent-cool");
    expect(btn).not.toHaveClass("usa-button--base");
  });

  it("adds appropriate base color class", () => {
    const { getByText } = render(
      <Button onClick={clickMock} color="base">
        testbtn
      </Button>
    );
    const btn = getByText("testbtn");
    expect(btn).toHaveClass("usa-button--base");
  });

  it("toggles focus class on mouse over", async () => {
    const { getByText } = render(<Button onClick={clickMock}>testbtn</Button>);
    const btn = getByText("testbtn");
    expect(btn).not.toHaveClass("usa-button--hover");
    fireEvent.mouseEnter(btn);
    await wait(() => {
      expect(btn).toHaveClass("usa-button--hover");
    });
    fireEvent.mouseLeave(btn);
    await wait(() => {
      expect(btn).not.toHaveClass("usa-button--hover");
    });
  });
});
