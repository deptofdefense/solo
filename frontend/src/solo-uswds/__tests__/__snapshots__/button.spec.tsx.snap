// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uswds button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="usa-button"
  >
    testbtn
  </button>
</DocumentFragment>
`;
